packet_injector/inject_packet.go

```diff
@@ -202,26 +209,40 @@ func InjectPacket(pp *PacketParams, g *graph.Graph) (string, error) {
 	f := flow.NewFlow()
 	f.InitFromGoPacket(flowKey, common.UnixMillis(time.Now()), &packet, int64(len(packetData)), tid, flow.FlowUUIDs{}, flow.FlowOpts{})
 
-	go func() {
+	p := make(chan bool)
+	chnl.Lock()
+	chnl.Pipes[pp.UUID] = p
+	chnl.Unlock()
+
+	go func(c chan bool) {
 		defer rawSocket.Close()
 
 		for i := int64(0); i < pp.Count; i++ {
-			logging.GetLogger().Debugf("Injecting packet on interface %s", ifName)
+			select {
+			case <-c:
+				logging.GetLogger().Debugf("Injection stoped on interface %s", ifName)
+				i = pp.Count
```

Comment:
I would prefer to have a break with a label, more readable

Reply:
done